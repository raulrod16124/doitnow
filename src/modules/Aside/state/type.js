export const asideTypes = {
  homeTodayView: "@home-today-view",
  homeTimestampView: "@home-timestamp-view",
};
