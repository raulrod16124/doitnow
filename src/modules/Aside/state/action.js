import { asideTypes } from "./type";

export const HomeViewsVisibility = (view) => {
  const typeofView = view
    ? asideTypes.homeTodayView
    : asideTypes.homeTimestampView;
  return {
    type: typeofView,
    payload: view,
  };
};
