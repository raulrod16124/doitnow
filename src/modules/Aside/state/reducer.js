import { asideTypes } from "./type";

export const AsideReducer = (state = false, action) => {
  switch (action.type) {
    case asideTypes.homeTodayView:
      state = action.payload;
      return state;
    case asideTypes.homeTimestampView:
      state = action.payload;
      return state;
    default:
      return state;
  }
};
