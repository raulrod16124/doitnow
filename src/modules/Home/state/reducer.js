import TODOS from './../provider/todos.json';
import { TYPES } from "./type";

const initialState = [
    {
        id: "1111",
        title: "BOOM",
        level: "easy",
        status:"todo",
        description: "nothing",
    }
]

export const TodosReducer = ( state = initialState, action ) => {
    
    switch (action.type) {
        case TYPES.getTodo:
            return state
        case TYPES.createTodo:
            return [...state, action.payload];
        case TYPES.updateTodo:
            console.log(action.payload.status);
            state[action.payload.id] = action.payload;
            return {...state};
        case TYPES.deleteTodo:
            delete state[action.payload];
            return {...state}; 
    
        default:
            return state;
    }
}