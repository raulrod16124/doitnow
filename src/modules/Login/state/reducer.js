import { TYPES } from "./type";

const initialState = {
  status: "initial",
  error: "",
  data: {},
};

export const LoginReducer = (state = initialState, action) => {
  switch (action.type) {
    case TYPES.checkUser:
      let userData = null;
      // // console.log(action.payload);
      if (action.payload.user?.stsTokenManager.refreshToken) {
        userData = {
          id: action.payload._tokenResponse.localId,
          token: action.payload._tokenResponse.idToken,
          email: action.payload.user.providerData[0].email,
          name: action.payload.user.providerData[0].email.split("@")[0],
        };
      }
      if (userData?.token) {
        return {
          ...state,
          data: userData,
          status: "success",
        };
      } else {
        return {
          ...state,
          error: action.payload.split("/")[1].replace("-", " "),
          status: "error",
        };
      }

    case TYPES.createNewUser:
      // console.log(action.payload);
      if (action.payload.email) {
        return {
          ...state,
          data: action.payload,
          status: "created_account",
        };
      } else {
        return {
          ...state,
          error: action.payload.split("/")[1].replace("-", " "),
          status: "error",
        };
      }
    // TODO - Refactor to use firebase
    case TYPES.updateUser:
    // // console.log(action.payload);
    // const updateTodo = state.map( todo => {
    //     if( todo.id === action.payload.id ){
    //         return todo = action.payload;
    //     };
    //     return todo;
    // } );
    // return updateTodo;
    case TYPES.deleteUser:
    // // console.log(action.payload);
    // const updateTodos = state.filter( todo => todo.id !== action.payload );
    // return updateTodos;

    default:
      return state;
  }
};
