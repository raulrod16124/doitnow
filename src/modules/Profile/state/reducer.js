import { TYPES } from "./type";

const initialState = {
  status: "initial",
  error: "",
  data: [],
};

export const ProfileReducer = (state = initialState, action) => {
  switch (action.type) {
    case TYPES.getUserProfile:
      console.log(action.payload);
      if (action.payload) {
        return {
          ...state,
          data: action.payload,
          status: "success",
        };
      } else {
        return {
          ...state,
          error: action.payload,
          status: "error",
        };
      }
    case TYPES.createUserProfile:
      console.log(action.payload);
      if (action.payload) {
        return {
          ...state,
          data: action.payload,
          status: "created profile success",
        };
      } else {
        return {
          ...state,
          error: action.payload,
          status: "error",
        };
      }
    case TYPES.updateUserProfile:
      if (action.payload) {
        return {
          ...state,
          status: "userData_updated",
        };
      } else {
        return {
          ...state,
          error: action.payload,
          status: "error",
        };
      }

    default:
      return state;
  }
};
