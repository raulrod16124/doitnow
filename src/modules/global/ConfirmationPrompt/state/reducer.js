const initialState = {
  status: "close",
  acceptButton: "Accept",
  cancelButton: "Cancel",
  handleReject: null,
  handleAccept: null,
  message: null,
};

export const ConfirmationPromptReducer = (state = initialState, action) => {
  switch (action.type) {
    case "@open_confirmation_propmt":
      // console.log("OPEN POPUP");
      return { ...state, status: "open", ...action.payload };
    case "@close_confirmation_propmt":
      // console.log("CLOSE POPUP");
      return initialState;
    default:
      return state;
  }
};
