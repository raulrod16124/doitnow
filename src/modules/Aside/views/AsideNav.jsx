import React, { useEffect, useState } from "react";
import { useDispatch } from "react-redux";
import { useLocation } from "react-router";
import { Link } from "react-router-dom";

import { Footer } from "../../global/Footer";
import { HomeViewsVisibility } from "../state/action";

export const AsideNav = () => {
  const dispatch = useDispatch();

  const location = useLocation();

  const initialSectionView = {
    home: true,
    stats: false,
    profile: false,
  };

  const [sectionView, setSectionView] = useState(initialSectionView);

  let classNames = require("classnames");
  let homeSelected = classNames("section", {
    selected: location.pathname === "/home",
  });
  let statsSelected = classNames("section", {
    selected: location.pathname === "/stats",
  });
  let profileSelected = classNames("section", {
    selected: location.pathname === "/profile",
  });

  // home view selector
  const [homeView, setHomeView] = useState(false);
  let todayViewClass = classNames("view", { selected: !homeView });
  let timestampViewClass = classNames("view", { selected: homeView });

  const handleShowHomeViews = (view) => {
    dispatch(HomeViewsVisibility(view));
    setHomeView(view);
  };

  return (
    <div className="aside">
      <div className="nav-bar">
        <div className="logo-app">
          <h1>DOITNOW</h1>
        </div>
        <div className="navegation-content">
          <div className={homeSelected}>
            <li
              className="section-content"
              onClick={() =>
                setSectionView({
                  ...sectionView,
                  home: !sectionView.home,
                })
              }
            >
              <Link to="/home" className="link">
                <i className="fas fa-home icon"></i>
                <p className="text">Home</p>
              </Link>
              {sectionView.home ? (
                <i className="fas fa-chevron-up icon arrow"></i>
              ) : (
                <i className="fas fa-chevron-down icon arrow"></i>
              )}
            </li>
            {sectionView.home && (
              <ul className="section-selector">
                <Link to="/home" className="link">
                  <li
                    className={todayViewClass}
                    onClick={() => handleShowHomeViews(false)}
                  >
                    <p className="view-text">
                      <i className="fas fa-tasks icon"></i> Today
                    </p>
                  </li>
                  <li
                    className={timestampViewClass}
                    onClick={() => handleShowHomeViews(true)}
                  >
                    <p className="view-text">
                      <i className="far fa-calendar-alt icon"></i> Calendar
                    </p>
                  </li>
                </Link>
              </ul>
            )}
          </div>
          <div className={statsSelected}>
            <li
              className="section-content"
              onClick={() =>
                setSectionView({ ...sectionView, stats: !sectionView.stats })
              }
            >
              <Link to="/stats" className="link">
                <i className="fas fa-chart-bar icon"></i>
                <p className="text">Stats</p>
              </Link>
              {sectionView.stats ? (
                <i className="fas fa-chevron-up icon arrow"></i>
              ) : (
                <i className="fas fa-chevron-down icon arrow"></i>
              )}
            </li>
            {sectionView.stats && (
              <ul className="section-selector">
                <Link to="/stats" className="link">
                  <li className="view">
                    <p className="view-text">
                      <i className="fas fa-chart-pie icon"></i> Global
                    </p>
                  </li>
                </Link>
              </ul>
            )}
          </div>
          <div className={profileSelected}>
            <li
              className="section-content"
              onClick={() =>
                setSectionView({
                  ...sectionView,
                  profile: !sectionView.profile,
                })
              }
            >
              <Link to="/profile" className="link">
                <i className="far fa-user-circle icon"></i>
                <p className="text">Profile</p>
              </Link>
              {sectionView.profile ? (
                <i className="fas fa-chevron-up icon arrow"></i>
              ) : (
                <i className="fas fa-chevron-down icon arrow"></i>
              )}
            </li>
            {sectionView.profile && (
              <ul className="section-selector">
                <Link to="/profile" className="link">
                  <li className="view">
                    <p className="view-text">
                      <i className="fas fa-user icon"></i> Account
                    </p>
                  </li>
                  <li className="view">
                    <p className="view-text">
                      <i className="fas fa-cog icon"></i> Settings
                    </p>
                  </li>
                </Link>
              </ul>
            )}
          </div>
        </div>
      </div>
      <Footer />
    </div>
  );
};
