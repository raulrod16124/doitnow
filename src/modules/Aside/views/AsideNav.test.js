import { render, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import { MemoryRouter, Route, Switch } from "react-router-dom";
import { applyMiddleware, combineReducers, createStore } from "redux";
import thunk from "redux-thunk";

import { AsideReducer } from "../state/reducer";
import { AsideNav } from "./AsideNav";

describe("Testing AsideNav component", () => {
  let mockStore;
  let wrapper;

  beforeEach(() => {
    const reducers = combineReducers({
      AsideReducer: AsideReducer,
    });
    mockStore = createStore(reducers, applyMiddleware(thunk));
    wrapper = render(
      <Provider store={mockStore}>
        <MemoryRouter>
          <AsideNav />
        </MemoryRouter>
      </Provider>
    );
  });

  test("should render AsideNav component correclty", () => {
    expect(wrapper).toMatchSnapshot();
  });

  // Add text to detect the App Logo

  test("should find a img Role", () => {
    expect(screen.getByRole("img")).toBeInTheDocument();
  });

  test("should find a Home text", () => {
    expect(screen.getByText("Home")).toBeInTheDocument();
  });

  test("should find a Today text", () => {
    expect(screen.getByText("Today")).toBeInTheDocument();
  });

  test("should find a Calendar text", () => {
    expect(screen.getByText("Calendar")).toBeInTheDocument();
  });

  test("should find a Profile text", () => {
    expect(screen.getByText("Profile")).toBeInTheDocument();
  });

  test("should find a Account text", () => {
    expect(screen.getByText("Account")).toBeInTheDocument();
  });

  test("should find a Disconnect text", () => {
    expect(screen.getByText("Disconnect")).toBeInTheDocument();
  });
});

describe("Testing functionality of AsideNav component", () => {
  let mockStore;
  let wrapper;

  let testHistory;

  beforeEach(() => {
    const reducers = combineReducers({
      AsideReducer: AsideReducer,
    });
    mockStore = createStore(reducers, applyMiddleware(thunk));
    wrapper = render(
      <Provider store={mockStore}>
        <MemoryRouter>
          <AsideNav />
          <Switch>
            <Route
              path={`*`}
              render={(routeProps) => {
                testHistory = routeProps.history;
                return null;
              }}
            />
          </Switch>
        </MemoryRouter>
      </Provider>
    );
  });

  test("should find a / path as an initial path", () => {
    // console.log(testHistory.location.pathname);
    expect(testHistory.location.pathname).toEqual("/");
  });
});
