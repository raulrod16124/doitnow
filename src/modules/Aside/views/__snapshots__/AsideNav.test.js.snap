// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing AsideNav component should render AsideNav component correclty 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="aside"
      >
        <div
          class="content-responsive-menu"
        >
          <div
            class="container-responsive"
          >
            <i
              class="fas fa-bars icon icon-menu"
            />
            <h1
              class="title-responsive"
            >
              DOITNOW
            </h1>
          </div>
        </div>
        <div
          class="bg-nav-bar"
        >
          <div
            class="nav-bar"
          >
            <i
              class="fas fa-times icon icon-close-nav"
            />
            <div
              class="logo-app"
            >
              <h1
                class="title"
              >
                DOITNOW
              </h1>
            </div>
            <div
              class="nav-avatar-content"
            >
              <div
                class="avatar-header"
              >
                <div
                  class="avatar-content"
                >
                  <p
                    class="user-name-content"
                  >
                    Hi
                     
                    <span
                      class="user-name"
                    >
                      
                      !!
                    </span>
                  </p>
                  <a
                    href="/profile"
                  >
                    <img
                      alt="avatar"
                      class="avatar"
                      src="https://cdn-icons-png.flaticon.com/128/616/616412.png"
                    />
                  </a>
                  <p
                    class="user-name-content-responsive"
                  >
                    Hi
                     
                    <span
                      class="user-name-responsive"
                    >
                      
                      !!
                    </span>
                  </p>
                </div>
              </div>
            </div>
            <div
              class="navegation-content"
            >
              <div
                class="section"
              >
                <li
                  class="section-content"
                >
                  <i
                    class="fas fa-home icon-section"
                  />
                  <p
                    class="text"
                  >
                    Home
                  </p>
                </li>
                <ul
                  class="section-selector"
                >
                  <a
                    class="link"
                    href="/home"
                  >
                    <li
                      class="view selected"
                    >
                      <p
                        class="view-text"
                      >
                        <i
                          class="fas fa-tasks icon"
                        />
                         Today
                      </p>
                    </li>
                    <li
                      class="view"
                    >
                      <p
                        class="view-text"
                      >
                        <i
                          class="far fa-calendar-alt icon"
                        />
                         Calendar
                      </p>
                    </li>
                  </a>
                </ul>
              </div>
              <div
                class="section"
              >
                <li
                  class="section-content"
                >
                  <i
                    class="far fa-user-circle icon-section"
                  />
                  <p
                    class="text"
                  >
                    Profile
                  </p>
                </li>
                <ul
                  class="section-selector"
                >
                  <a
                    class="link"
                    href="/profile"
                  >
                    <li
                      class="view"
                    >
                      <p
                        class="view-text"
                      >
                        <i
                          class="fas fa-user icon"
                        />
                         Account
                      </p>
                    </li>
                  </a>
                </ul>
              </div>
            </div>
            <div
              class="content-logout"
            >
              <footer
                class="footer"
              >
                <p
                  class="footer-text"
                >
                  <i
                    class="fas fa-sign-out-alt"
                  />
                   Disconnect
                </p>
              </footer>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="aside"
    >
      <div
        class="content-responsive-menu"
      >
        <div
          class="container-responsive"
        >
          <i
            class="fas fa-bars icon icon-menu"
          />
          <h1
            class="title-responsive"
          >
            DOITNOW
          </h1>
        </div>
      </div>
      <div
        class="bg-nav-bar"
      >
        <div
          class="nav-bar"
        >
          <i
            class="fas fa-times icon icon-close-nav"
          />
          <div
            class="logo-app"
          >
            <h1
              class="title"
            >
              DOITNOW
            </h1>
          </div>
          <div
            class="nav-avatar-content"
          >
            <div
              class="avatar-header"
            >
              <div
                class="avatar-content"
              >
                <p
                  class="user-name-content"
                >
                  Hi
                   
                  <span
                    class="user-name"
                  >
                    
                    !!
                  </span>
                </p>
                <a
                  href="/profile"
                >
                  <img
                    alt="avatar"
                    class="avatar"
                    src="https://cdn-icons-png.flaticon.com/128/616/616412.png"
                  />
                </a>
                <p
                  class="user-name-content-responsive"
                >
                  Hi
                   
                  <span
                    class="user-name-responsive"
                  >
                    
                    !!
                  </span>
                </p>
              </div>
            </div>
          </div>
          <div
            class="navegation-content"
          >
            <div
              class="section"
            >
              <li
                class="section-content"
              >
                <i
                  class="fas fa-home icon-section"
                />
                <p
                  class="text"
                >
                  Home
                </p>
              </li>
              <ul
                class="section-selector"
              >
                <a
                  class="link"
                  href="/home"
                >
                  <li
                    class="view selected"
                  >
                    <p
                      class="view-text"
                    >
                      <i
                        class="fas fa-tasks icon"
                      />
                       Today
                    </p>
                  </li>
                  <li
                    class="view"
                  >
                    <p
                      class="view-text"
                    >
                      <i
                        class="far fa-calendar-alt icon"
                      />
                       Calendar
                    </p>
                  </li>
                </a>
              </ul>
            </div>
            <div
              class="section"
            >
              <li
                class="section-content"
              >
                <i
                  class="far fa-user-circle icon-section"
                />
                <p
                  class="text"
                >
                  Profile
                </p>
              </li>
              <ul
                class="section-selector"
              >
                <a
                  class="link"
                  href="/profile"
                >
                  <li
                    class="view"
                  >
                    <p
                      class="view-text"
                    >
                      <i
                        class="fas fa-user icon"
                      />
                       Account
                    </p>
                  </li>
                </a>
              </ul>
            </div>
          </div>
          <div
            class="content-logout"
          >
            <footer
              class="footer"
            >
              <p
                class="footer-text"
              >
                <i
                  class="fas fa-sign-out-alt"
                />
                 Disconnect
              </p>
            </footer>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
